r-pkg/R/elasticsearch_eda_funs.R

```diff
@@ -114,3 +114,93 @@ get_counts <- function(field
     
 }
 
+#' @title Get the names and data types of the indexed fields in an index
+#' @name get_fields
+#' @description For a given Elasticsearch index, return the mapping from field name
+#'              to data type for all indexed fields.
+#' @importFrom futile.logger flog.fatal flog.info
+#' @importFrom httr GET content stop_for_status
+#' @param es_host A string identifying an Elasticsearch host. This should be of
+#'                the form \code{[transfer_protocol][hostname]:[port]}. For example,
+#'                \code{'http://myindex.thing.com:9200'}.
+#' @param es_indexes A character vector that contains the names of indexes for
+#'                   which to get mappings. Default, is \code{'_all'}, which means
+#'                   get the mapping for all indexes. Names of indexes can be
+#'                   treated as regular expressions.
+#' @export
+#' @return A data.table containing four columns: index, type, field, and data_type
+#' @examples \dontrun{
+#' # get the mapping for all indexed fields in the ticket_sales and customers indexes
+#' mappingDT <- retrieve_mapping(es_host = "http://es.custdb.mycompany.com:9200"
+#'                               , es_indexes = c("ticket_sales", "indexes"))
+#' }
+get_fields <- function(es_host
+                       , es_indexes = '_all'
+) {
+    
+    # Input checking
+    url <- .ValidateAndFormatHost(es_host)
+    
+    # collapse character vectors into comma separated strings. If any arguments
+    # are NULL, create an empty string
+    indexes <- paste(es_indexes, collapse = ',')
+    
+    ########################## build the query ################################
+    if (nchar(indexes) > 0) {
+        url <- paste(url, indexes, '_mapping', sep = '/')
+    } else {
+        msg <- paste("get_fields must be passed a valid es_indexes."
+                     , "You provided", paste(es_indexes, collapse = ', ')
+                     , 'which resulted in an empty string')
+        futile.logger::flog.fatal(msg)
+        stop(msg)
+    }
+    
+    ########################## make the query ################################
+    futile.logger::flog.info(paste('Getting indexed fields for indexes:', indexes))
+    
+    result <- httr::GET(url = url)
+    httr::stop_for_status(result)
+    resultContent <- httr::content(result)
+    
+    ##################### return the flattened result #########################
+    return(.flatten_mapping(mapping = resultContent))
```

Comment:
Thanks. I had to do that in order to unit test. and then i realized that this was probably a better design anyway.